app/bundles/ReportBundle/Model/ReportModel.php

```diff
@@ -47,6 +47,11 @@ class ReportModel extends FormModel
 {
     const CHANNEL_FEATURE = 'reporting';
 
+    /**
+     * @var array
+     */
+    public $reportBuilderData;
```

Comment:
I understand you made this public to make the test easier, but we should not make properties public only for tests. I like that you replaced the static method variable with a class property. That's more transparent.
```suggestion
    private $reportBuilderData;
```

Let's emulate a subscriber and set the data to the dispatched event instead. Here is an example:
https://github.com/mautic/mautic/blob/3.x/app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php#L383-L396